### PR TITLE
Enhance/5278 - Redirect to OAuth flow from the Setup Component to provide the edit scope when needed

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue:

- #5278 

## Relevant technical choices

The below aspects of the IB weren't needed due to the existing functionality of the permissions modal:

> - Get the permissions error using the `getPermissionScopeError` selector of `CORE_USER` store.
> - Get the connect URL to navigate using the `getConnectURL` selector of the `CORE_USER` store.
> - Pass an object to the `getConnectURL` selector with the following properties:
>   * `additionalScopes` with the value `scopes` obtained from the `getPermissionScopeError` selector.
>   * `redirectURL` with value `global.location.href`.
> - Modify the `handleCTAClick` to async callback using the `useCallback` hook and add the following:
>   * Call the `snapshotAllStores` function from the `assets/js/googlesitekit/data/create-snapshot-store.js` and pass the `registry` using the `useRegistry` hook.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

**Review by techanvil — CHANGES_REQUESTED** [review]
Hey @hussain-t, this is looking good. As discussed, please can you update it so the footer text shows the permissions message for the "existing tag" permutation as well.

Additionally I'll create a followup issue to address the "existing property" variants that were missed in the AC.

File: assets/js/modules/analytics-4/constants.js [hunk 1]
```diff
@@ -22,3 +22,5 @@ export const ACTIVATION_STEP_SUCCESS = 2;
 
 export const ACTIVATION_ACKNOWLEDGEMENT_TOOLTIP_STATE_KEY =
 	'activation-acknowledgement-tooltip-state';
+
+export const GA4_ACTIVATION_BANNER_STATE = 'ga4-activation-banner-state';
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/js/modules/analytics-4/constants.js]: Updated.

**Review by techanvil — CHANGES_REQUESTED** [review]
@hussain-t my apologies for the back and forth here but as discussed on Slack we do need to revisit this and redirect manually to the OAuth flow, etc.

File: assets/sass/components/analytics-4/_googlesitekit-analytics-4-setup-banner.scss [hunk 2]
```diff
@@ -45,4 +45,8 @@
 			row-gap: 24px;
 		}
 	}
+
+	.googlesitekit-ga4-setup-banner__footer-text {
```
[the remark this answers is not in the document]
Comment (hussain-t) [reply assets/sass/components/analytics-4/_googlesitekit-analytics-4-setup-banner.scss:49]: Updated.

**Comment by hussain-t** [comment]
> @hussain-t my apologies for the back and forth here but as discussed on Slack we do need to revisit this and redirect manually to the OAuth flow, etc.

No problem, @techanvil. I have updated the changes.

**Review by techanvil — APPROVED** [review]
LGTM! Nice work @hussain-t

**Pull request merged**